src/test/java/seedu/address/model/position/UniquePositionListTest.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    @Test
+    public void setPosition_editedPersonHasDifferentIdentity_success() {
+        uniquePositionList.add(ADMIN_ASSISTANT);
+        uniquePositionList.setPosition(ADMIN_ASSISTANT, BOOKKEEPER);
+        UniquePositionList expectedUniquePositionList = new UniquePositionList();
+        expectedUniquePositionList.add(BOOKKEEPER);
+        assertEquals(expectedUniquePositionList, uniquePositionList);
+    }
+
+    @Test
+    public void setPosition_editedPersonHasNonUniqueIdentity_throwsDuplicatePositionException() {
+        uniquePositionList.add(ADMIN_ASSISTANT);
+        uniquePositionList.add(BOOKKEEPER);
+        assertThrows(DuplicatePositionException.class, () ->
+                uniquePositionList.setPosition(ADMIN_ASSISTANT, BOOKKEEPER));
+    }
+
+    @Test
+    public void remove_nullPerson_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniquePositionList.remove(null));
+    }
+
+    @Test
+    public void remove_positionDoesNotExist_throwsPositionNotFoundException() {
+        assertThrows(PositionNotFoundException.class, () -> uniquePositionList.remove(ADMIN_ASSISTANT));
+    }
+
+    @Test
+    public void remove_existingPerson_removesPerson() {
+        uniquePositionList.add(ADMIN_ASSISTANT);
+        uniquePositionList.remove(ADMIN_ASSISTANT);
+        UniquePositionList expectedUniquePositionList = new UniquePositionList();
+        assertEquals(expectedUniquePositionList, uniquePositionList);
+    }
+
+    @Test
+    public void setPositions_nullUniquePositionList_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniquePositionList.setPositions((UniquePositionList) null));
+    }
+
+    @Test
+    public void setPositions_uniquePositionList_replacesOwnListWithProvidedUniquePositionList() {
+        uniquePositionList.add(ADMIN_ASSISTANT);
+        UniquePositionList expectedUniquePositionList = new UniquePositionList();
+        expectedUniquePositionList.add(BOOKKEEPER);
+        uniquePositionList.setPositions(expectedUniquePositionList);
+        assertEquals(expectedUniquePositionList, uniquePositionList);
+    }
+
+    @Test
+    public void setPositions_nullList_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniquePositionList.setPositions((List<Position>) null));
+    }
+
+    @Test
+    public void setPositions_list_replacesOwnListWithProvidedList() {
+        uniquePositionList.add(ADMIN_ASSISTANT);
+        List<Position> positionList = Collections.singletonList(BOOKKEEPER);
+        uniquePositionList.setPositions(positionList);
+        UniquePositionList expectedUniquePositionList = new UniquePositionList();
+        expectedUniquePositionList.add(BOOKKEEPER);
+        assertEquals(expectedUniquePositionList, uniquePositionList);
+    }
+
+    @Test
+    public void setPositions_listWithDuplicatePersons_throwsDuplicatePositionException() {
```

Comment:
perhaps "person" should be "position" here?

Reply:
Changes made accoordinly.